packages/sdk-v2/src/sdk/claim/react.ts

```diff
@@ -33,6 +37,105 @@ export const useIsAddressVerified = (address: string, env?: EnvKey) => {
+  // all contracts to claim from
+  const [poolContracts, setPoolContracts] = useState<Contract[]>([]);
+  // all contracts that have been claimed from
+  const [claimedContracts, setClaimedContracts] = useState<Contract[]>([]);
+
+  const { state, send, resetState } = useContractFunctionWithDefaultGasFees(contract, "claim", {
+    transactionName: "Claimed UBI"
+  });
+
+  const nextClaim = useCallback(async () => {
+    if (!contract) return;
+
+    await send();
+
+    setClaimedContracts((prev: Contract[]) => [contract, ...prev]);
+    const remainingContracts = poolContracts.find(c => !claimedContracts.includes(c) && c !== contract);
+    setContract(remainingContracts);
+  }, [contract, poolContracts]);
+
+  useEffect(() => {
+    if (poolsDetails.length && !poolContracts.length) {
+      const poolContracts = getContractsFromClaimPools(poolsDetails);
+      setPoolContracts(poolContracts);
+    }
+  }, [poolsDetails]);
+
+  useEffect(() => {
+    if (contract) {
+      void nextClaim();
+    } else {
+      setClaimedContracts([]);
+    }
+
+    return () => resetState();
+  }, [contract]);
+
+  return {
+    poolContracts,
+    claimFlowStatus: {
+      isClaimingDone: claimedContracts.length === poolContracts.length,
+      remainingClaims: poolContracts.length - claimedContracts.length
+    },
+    setContract,
+    setPoolContracts,
+    setClaimedContracts,
+    transactionState: { state, resetState }
+  };
+};
+
+export const useGetMemberUBIPools = () => {
+  const { account, library, chainId } = useEthers();
+  const [poolsDetails, setPoolsDetails] = useState<PoolDetails[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+  const factory = GoodCollectiveContracts["42220"]?.find(envs => envs.name === "development-celo")?.contracts
+    .UBIPoolFactory;
+  const pool = GoodCollectiveContracts["42220"]?.find(envs => envs.name === "development-celo")?.contracts.UBIPool;
+
+  const fetchPools = useCallback(async () => {
+    setLoading(true);
+    setError(null);
+
+    try {
+      if (!library || !account || !factory || !pool || chainId !== 42220) {
+        setPoolsDetails([]);
+        return;
+      }
+
+      const poolAddresses = await getMemberPools(account, factory, library);
+
+      if (!poolAddresses || !poolAddresses.length) {
+        setPoolsDetails([]);
+        return;
+      }
+
+      const details = await getPoolsDetails(poolAddresses, pool, library, account);
+
+      setPoolsDetails(details);
+    } catch (err: any) {
+      console.error("Error fetching pools:", err);
+      setError(err.message);
+    } finally {
+      setLoading(false);
+    }
+  }, [account, library, chainId]);
+
+  useEffect(() => {
+    void fetchPools();
+  }, [fetchPools]);
+
+  return useMemo(
+    () => ({ poolsDetails, loading, error, fetchPools }),
+    [account, poolsDetails, factory, pool, loading, error]
+  );
```

Comment:
usememo is redundant here

Reply:
removed: https://github.com/GoodDollar/GoodWeb3-Mono/pull/162/commits/79657860d0f7efa5a7149908f9b0b64b85ef6770